### PR TITLE
Fix #89 Hooks cannot remove inherited properties

### LINKED ISSUE #89: Hooks cannot remove inherited properties
I use `feathers-mongoose` and try to remove `password` field with `after hook` but it doesn't work. 

I found that it's caused by [containsField() and removeField() method](https://github.com/feathersjs/feathers-hooks/blob/db68ef7f877ed92008f678e97f8ef58f573737ec/src/bundled.js#L102-L119) that use `hasOwnProperty()` to check for property existence. However, properties in an object from `feathers-mongoose` are inherited properties which `hasOwnProperty()` will return `false`. 

I will make an PR to solve this issue by searching through the prototype chain to remove inherited properties.

### PR DESCRIPTION
**Pull request by supasate**
Fix #89 issue

### CODE REVIEW

File: src/bundled.js [hunk 1]
```diff
@@ -99,39 +99,46 @@ export function pluckQuery(... fields) {
...
 }
 
-function containsField(obj, find) {
-    var args = find.split('.');
-
-  for (var i = 0; i < args.length; i++) {
-    if (!obj || !obj.hasOwnProperty(args[i])) {
-      return false;
+function findFieldOwner(obj, field) {
+  const nestedFields = field.split('.');
```
Comment (daffl) [inline src/bundled.js]: Isn't this just a longer way of saying `!obj || !obj[args[i]]`?

Comment (supasate) [inline src/bundled.js]: If using `typeof obj[args[i]] === undefined`, we will not know whether the property is belong to the current `obj` or ancestor prototype (direct owner). To delete that property, we need to delete it from the direct owner. So, we need to use `hasOwnProperty`.

(About the const name `nestedFields`, I just make it more explicit but I can just use `args[i]` if you think it's more concise.)

Comment (supasate) [inline src/bundled.js]: I refactored the code in my new commit to make it looks cleaner and more concise. Plz take a look. (I'll squash it later).

**Comment by daffl** [comment]
Are you sure that `delete` only works on the actual object? I just tried:

```js
const o = { test: 'me' }
const oo = Object.create(o);

delete oo.test;

console.log(oo);
```

And I am getting an empty object logged.

**Comment by supasate** [comment]
Yes. I'm sure. It's hidden in `__proto__`.
image

**Comment by supasate** [comment]
It's the effect of prototype inheritance. 

>  If the delete operator succeeds, it removes the property from the object entirely. However, if a property with the same name exists on the object's prototype chain, the object will inherit that property from the prototype.

[Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete#Description)

**Comment by daffl** [comment]
Done. Released in v1.5.6. Thank you for bearing with me 😄

**Comment by supasate** [comment]
Never mind @daffl. It took me several days before figuring it out too!

**Pull request merged**